### PR TITLE
hold for pre-review all Promoted addons in applicable groups, not just recommended

### LINKED ISSUE #14862: Pre-review for Promoted addons
> When an add-on is added to either of the Verified, Spotlight, or Line lists, it won’t be fully added until its next version passes review, like it’s done with Recommended Add-ons.

> [P2] Make auto-approval a configuration option per list.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #14862

### CODE REVIEW

File: src/olympia/discovery/models.py [hunk 1]
```diff
@@ -11,10 +11,6 @@
 
 
 class DiscoveryItem(OnChangeMixin, ModelBase):
-    RECOMMENDED = 'Recommended'
-    PENDING_RECOMMENDATION = 'Pending Recommendation'
-    NOT_RECOMMENDED = 'Not Recommended'
-
```
Comment (eviljeff) [inline src/olympia/discovery/models.py:17]: drive-by clean-up of some now unused constants.

File: src/olympia/reviewers/migrations/0007_auto_20200730_1335.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    dependencies = [
+        ('reviewers', '0006_auto_20200609_1655'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='autoapprovalsummary',
+            name='is_promoted_prereview',
+            field=models.BooleanField(default=False, help_text='Is in a promoted addon group that requires pre-review'),
```
Comment (diox) [inline src/olympia/reviewers/migrations/0007_auto_20200730_1335.py:16]: That won't do it. Here and in the model you need a `NullBooleanField` (or `BooleanField` with `null=True`) for backwards-compatibility with previous code. Otherwise there won't be a database-level default (*) and code still running the old tag will fail.

(*) When migrating, django generates one to set a value in the column for every row, and then immediately throws it away, considering that you're only running new code, which isn't the case. This is also the reason why I prefer to add those new fields with `default=None` - this avoid the triple `ALTER` dance django is doing with the database default - we can then later fix the data by changing the default to `False` and then at a later push remove the `null=True` + add a migration to convert `None` to `False`.

Comment (eviljeff) [reply src/olympia/reviewers/migrations/0007_auto_20200730_1335.py]: https://github.com/mozilla/addons-server/pull/15122/commits/7cbd7eade4fb5a4ea2f39360c2dfea5cf6fa8a22

Comment (diox) [reply src/olympia/reviewers/migrations/0007_auto_20200730_1335.py]: Please change the `default` as well to avoid multiple unnecessary `ALTER TABLE`

Comment (eviljeff) [reply src/olympia/reviewers/migrations/0007_auto_20200730_1335.py]: https://github.com/mozilla/addons-server/pull/15122/commits/5e388c98e17677ff71f502cd1d64e4c2394410b4 but we need https://github.com/mozilla/addons-server/issues/12334 to be done really.
